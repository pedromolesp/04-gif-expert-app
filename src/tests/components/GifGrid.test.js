const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
const { useFetchGifs } = require("../../hooks/useFetchGifs");

jest.mock('../../hooks/useFetchGifs');
const category = "One punch";
describe('Pruebas en <GifGrid/>', () => {
    test('debe mostrar el componente corretamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid key={category} category={category} />);
        expect(wrapper).toMatchSnapshot();

    });
    test('debe mostrar items cuando se cargar imÃ¡genes', () => {
        const gifs = [
            { id: 23, title: "ole", url: "erger" }
        ]
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });
        const wrapper = shallow(<GifGrid key={category} category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
    });

});