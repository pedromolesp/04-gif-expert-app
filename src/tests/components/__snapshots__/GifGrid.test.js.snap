// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe mostrar el componente corretamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
